### ISSUE TITLE
Rfsrc curve is overfitting quite a bit. What could be the reasons?

### ISSUE THREAD

**Issue by madhurgupta27** [kogalur/randomForestSRC#42]
[test_data.zip](https://github.com/kogalur/randomForestSRC/files/3417288/test_data.zip)

Hello Sir, 

I have ran RFSRC on my train data and scored on the test data. While the model is performing very well on the train data, it is showing huge overfitting on test data. I thought that maybe presenting the problem to you might be useful given your expert knowledge of how Survival Forests work. 
The data is attached with this mail.
I am first running the below code to make the model. 
```R
library(randomForestSRC)

# Running the model
[train_data.zip](https://github.com/kogalur/randomForestSRC/files/3417313/train_data.zip)

rfsrc_model <- rfsrc(Surv(monthlap,islapsed) ~ .,
                     data=train, 
                     ntree = 100 , 
                     do.trace = TRUE)

# Making predictions on the test dataset
pred_test <- predict.rfsrc(rfsrc_model, test)

# Making the dataset containing 24 month probabilities and the time (monthlap) and status # # 
# (islapsed) variables for the test data
Test_Survival = as.data.frame(cbind(pred_test$survival, monthlap = test$monthlap, islapsed = test$islapsed))
```
Then, the way I'm checking the performance on test data is by estimating monthly survival probabilities of all people in the data for a time period of 24 months from the model, setting a cut-off probability, below which if the probability of any member goes for any month, he'll be considered dead in that particular month, and then drawing a graph with percentage of deaths per month in 24 months on the y-axis, and the time on the x-axis. This graph is drawn for both the train and test data and in my case, the test graph line is hugely overpredicting death percentage as compared to train percentage, particularly beyond the 6th month. I'm attaching one plot with this question. 
[Train_test_plot.pdf](https://github.com/kogalur/randomForestSRC/files/3417309/Train_test_plot.pdf)
This is as per the below code:
```R
# input the probability cut-off
x <- as.numeric(0.60)

# Plotting the graph
ncap_surv2 <- data.frame()
ncap_surv5 <- data.frame()
ncap_surv2 <- ifelse(ncap_surv[,c(1:24)]<x,1,0)
ncap_surv2 = cbind(ncap_surv2,monthlap = ncap_surv$monthlap, islapsed = ncap_surv$islapsed)
for (i in 24:2){
  ncap_surv2[, i] = ncap_surv2[, i] - ncap_surv2[, i-1]
}
ncap_surv2 <- as.data.frame(ncap_surv2)
ncap_surv3 <- ncap_surv2 %>% 
  group_by(monthlap, islapsed) %>% 
  summarise(n = n())
ncap_surv4 <- colSums(ncap_surv2[, c(1:24)])
ncap_surv5 <- cbind(ncap_surv4, monthlap = 1:24, islapsed = 1)
ncap_surv5 <- as.data.frame(ncap_surv5)
ncap_surv6 <- merge(x = ncap_surv3, y = ncap_surv5, by = c('monthlap', 'islapsed'), all.x = TRUE)
ncap_surv7 <- ncap_surv6[!(ncap_surv6$islapsed == 0), ]
colnames(ncap_surv7) <- c('monthlap', 'islapsed', 'actual', 'predicted')
ncap_surv7[, c("actual", "predicted")] <- ncap_surv7[, c("actual", "predicted")]/nrow(ncap_surv)
ncap_surv7 <- within(ncap_surv7, cum_actual <- cumsum(actual))
ncap_surv7 <- within(ncap_surv7, cum_predicted <- cumsum(predicted))

library(ggplot2)
p = ggplot() + 
  geom_line(data = ncap_surv7, aes(x = monthlap, y = cum_actual), color = "blue") +
  geom_line(data = ncap_surv7, aes(x = monthlap, y = cum_predicted), color = "red") +
  xlab('Month') +
  ylab('percent')
print(p)
```
I am totally at wits' end to ascertain why the survival probabilities are suddenly decreasing beyond the 6-month mark. Which things should I look into to ascertain which variable or other cause is decreasing the survival probabilities?

Thanks!

**Comment by madhurgupta27** [comment]
This is the train test data:

[train_data.zip](https://github.com/kogalur/randomForestSRC/files/3417322/train_data.zip)
[test_data.zip](https://github.com/kogalur/randomForestSRC/files/3417327/test_data.zip)

**Comment by kogalur** [comment]
We're not convinced that your example shows that there is a problem with the code.  You would need to provide a more simple example that clearly and convincingly demonstrates a problem with test set overfitting.
